### PR TITLE
Makefile and Dockerfiles for CPU/CUDA/Jetson

### PR DESCRIPTION
**Pull request by EmilianoHFlores**
Included Dockerfile templates for CPU, CUDA and Jetson (L4T 35), with ROS Noetic and user configuration installed. Included a Makefile for easier image and container creation and execution.

### CODE REVIEW

File: docker/scripts/build.bash [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+fi
+
+if [ -n "$JETSON_L4T" ]; then
+    DOCKER_FILE="$PWD/docker/Dockerfile.${AREA}.l4t-$JETSON_L4T"
+    echo "Building for Nvidia Jetson with L4T: $JETSON_L4T"
+fi
+
+echo "Building docker image: $DOCKER_FILE"
+
+docker build -t home-$AREA \
```
Comment (afr2903) [inline docker/scripts/build.bash:47]: For the containers it's okay having the same name for its ephimeral philosophy, but for the images, I think we should use `tags` for local handling of different images and maybe deployment in the future.

I suggest choosing the convention of `home/feature:version` or `home/area:version`.

To adapt the script, an additional variable `$VERSION` could be initialized as `"base"`, and inside the conditionals for CUDA and Jetson, change the value to `"cuda"` or `"jetson"`. The docker command should look like:
```bash
docker build -t home/$AREA:$VERSION ...
```

File: docker/scripts/run.bash [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    -e DISPLAY=$DISPLAY \
+    -v /tmp/.X11-unix:/tmp/.X11-unix \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -v /dev:/dev \
+    --device /dev/video0:/dev/video0 \
+    $VOLUME_COMMANDS \
+    -w /workspace \
+    --name=$CONTAINER_NAME \
+    $IMAGE_NAME \
+    bash
```
Comment (afr2903) [inline docker/scripts/run.bash:107]: Seems great. For the **HRI** area I found useful the parameter `--env-file` to pass enviroment variables such as API keys and credentials. But if we create a `Makefile` for each area, I don't think that feature should be included here.

File: README.md [hunk 3]
```diff
@@ -19,12 +19,59 @@ And to update the submodules content execute:
...
+
+You can stop and remove the container with:
+
+```bash
+make main.down
+make main.remove
+```
+
+Additional commands can be added within the Makefile and the scripts inside the `docker/scripts` folder can help for easier integration and sharing.
+
```
Comment (afr2903) [inline README.md:74]: Everyone should be encouraged to modifiy the `Dockerfiles` and `Makefiles` for any environment change, such as the installation of a new dependency or a different parameter needed for the `docker run`.
I think that recommendation should be explicitly added here, and being remarked at each meeting.

File: docker/Dockerfile.main.cuda [hunk 4]
```diff
@@ -0,0 +1,168 @@
+#############################################
+# Created from althack/ros:noetic
+#############################################
+
+###########################################
+# Base image 
+###########################################
+FROM nvidia/cuda:11.8.0-runtime-ubuntu20.04 AS base
+
+LABEL maintainer="Emiliano Flores <A00833160@tec.mx>"
```
Comment (afr2903) [inline docker/Dockerfile.main.cuda:10]: Please change it for `RoBorregos <roborregosteam@gmail.com>` or removed 🙏

**Review by afr2903 — APPROVED** [review]
I left a few comments with suggestions. Feel free to update for the ones you think are the most convenient and merge this pull request :)

File: docker/scripts/build.bash (hunk 1 above)
Comment (afr2903) [reply docker/scripts/build.bash:47]: After a little research and by looking at some examples, now I suggest our naming convention should be: `roborregos/home-area:version`, or even more generalized `roborregos/home:area-version`.

Let's discuss the naming conventions!

File: docker/scripts/run.bash (hunk 2 above)
Comment (EmilianoHFlores) [reply docker/scripts/run.bash:107]: Specified $ADDITIONAL_COMMANDS usage for extra commands specific to areas.

File: docker/scripts/build.bash (hunk 1 above)
Comment (EmilianoHFlores) [reply docker/scripts/build.bash:47]: I agree with using roborregos/home:area-version. I'll include version to be "cpu", "cuda" or "l4t-xx.yy.zz".

**Comment by EmilianoHFlores** [comment]
I've included all suggested changes and added [multi-stage builds](https://docs.docker.com/build/building/multi-stage/) which I believe will make integration of individual features easier within each area's dockerfiles.

**Review by afr2903 — APPROVED** [review]
Seems great the feature of the multi-stage build! We can now merge this changes for everyone to use in the areas.

**Pull request merged**